### DISCUSSION TITLE
Draw holes in Polygon

### DISCUSSION THREAD

**Discussion by PothpothBR** [mapsforge/mapsforge#1364] [Q&A] [answered]
Hi, I've been trying for a few days to find a way to draw polygons with holes, from what I found in old forums there was a way to do this in the past. And possibly there is a way to do that according to this post:
https://groups.google.com/g/mapsforge-dev/c/gjtNUBD9Eg4/m/h5wqT1PCAgAJ

Sorry if I'm being too layman on this point, but I haven't found the starting point for this. Note When drawing polygons as layers, if there are one inside the other, they will overlap.

![image](https://user-images.githubusercontent.com/41635420/197770005-771d4d80-0d14-4faf-95d5-2b515aaf9afa.png)

My use for this would be drawing polygons from postgis, according to the format:
`[[[points],[sub-points]], [[othr-point],[othr-sub]]]`

My question would be a way to draw these polygons with holes, even if using another form besides the `Polygon` class. I appreciate the help if possible.

**Comment by devemux86** [comment, marked as answer]
Set the polygons points:
- Shell (clockwise)
- Holes (counterclockwise)

(or vice versa)

```java
List<LatLong> latLongs = new ArrayList<>();
// Shell
latLongs.add(new LatLong(-8, -8));
latLongs.add(new LatLong(8, -8));
latLongs.add(new LatLong(8, 8));
latLongs.add(new LatLong(-8, 8));
latLongs.add(new LatLong(-8, -8));
// Hole 1
latLongs.add(new LatLong(2, -6));
latLongs.add(new LatLong(2, -2));
latLongs.add(new LatLong(6, -2));
latLongs.add(new LatLong(6, -6));
latLongs.add(new LatLong(2, -6));
// Hole 2
latLongs.add(new LatLong(-6, 2));
latLongs.add(new LatLong(-6, 6));
latLongs.add(new LatLong(-2, 6));
latLongs.add(new LatLong(-2, 2));
latLongs.add(new LatLong(-6, 2));
polygon.setPoints(latLongs);
```

![Mapsforge](https://user-images.githubusercontent.com/3484020/197782193-dff09316-ace3-4a1e-9efc-db67712dd775.png)

**Reply by PothpothBR** [reply]
thanks for the reply, this solves my problem
